packages/app-builder/src/components/Decisions/RulesDetail.tsx

```diff
@@ -1,116 +1,152 @@
+                        }}
+                        values={{
+                          score: formatNumber(rule.scoreModifier, {
+                            language,
+                            signDisplay: 'always',
+                          }),
+                        }}
+                      />
+                    </div>
+                    {rule.formula ? (
+                      <RuleFormula
+                        formula={rule.formula}
+                        databaseAccessors={databaseAccessors}
+                        payloadAccessors={payloadAccessors}
+                        astOperators={astOperators}
+                        dataModel={dataModel}
+                        customLists={customLists}
+                        triggerObjectType={triggerObjectType}
+                      />
+                    ) : null}
+                  </div>
+                </Accordion.Content>
+              </Accordion.Item>
+            );
+          })}
         </Accordion.Container>
       </Collapsible.Content>
     </Collapsible.Container>
   );
-};
-
-const RuleSection = ({
-  rule,
-  value,
-}: {
-  rule: Decision['rules'][0];
-  value: string;
-}) => {
-  const status = rule.result ? 'triggered' : rule.error ? 'error' : undefined;
-
-  if (rule.description) {
-    return <RuleWithDescription rule={rule} status={status} value={value} />;
-  }
+}
 
-  return <RuleDetail rule={rule} status={status} />;
-};
-
-const RuleWithDescription = ({
-  rule,
-  status,
-  value,
+function RuleFormula({
+  formula,
+  databaseAccessors,
+  payloadAccessors,
+  astOperators,
+  dataModel,
+  customLists,
+  triggerObjectType,
 }: {
-  rule: Decision['rules'][0];
-  status?: RuleStatus;
-  value: string;
-}) => {
-  const { t } = useTranslation(decisionsI18n);
-
-  return (
-    <Accordion.Item value={value}>
-      <Accordion.Title className="flex flex-1 items-center justify-between gap-4">
-        <RuleDetail rule={rule} status={status} />
-        <Accordion.Arrow />
-      </Accordion.Title>
-      <Accordion.Content>
-        <div className="bg-purple-10 mt-4 flex gap-2 rounded p-4 text-purple-100">
-          <Icon icon="tip" className="size-5" />
-          <div className="flex flex-col gap-2">
-            <div className="font-semibold">
-              {t('decisions:rules.description')}
-            </div>
-            <div>{rule.description}</div>
-          </div>
-        </div>
-      </Accordion.Content>
-    </Accordion.Item>
-  );
-};
+  formula: AstNode;
+  databaseAccessors: DatabaseAccessAstNode[];
+  payloadAccessors: PayloadAstNode[];
+  astOperators: AstOperator[];
+  dataModel: TableModel[];
+  customLists: CustomList[];
+  triggerObjectType: string;
+}) {
+  const astEditor = useAstBuilder({
+    backendAst: formula,
+    localValidation: null,
+    databaseAccessors,
+    payloadAccessors,
+    astOperators,
+    dataModel,
+    customLists,
+    triggerObjectType,
+    onValidate: () => {},
+  });
```

Comment:
This still feels wrong to use a "edit" specific hook in order to display the AST (ex: we need to feed `onValidate: () => {}`)

There might be a better abstraction to handle the whole thing but :
- this work like this (so let's move on)
- I prefer to wait for the next steps in order to find a better abstraction (= inject intermediate evaluation errors)

Reply:
yes totally agreed